### PR TITLE
[mpact][passes] Setup MPACT python pipeline interface.

### PR DESCRIPTION
**Pull request by PeimingLiu**

### CODE REVIEW

File: python/CMakeLists.txt [hunk 1]
```diff
@@ -2,28 +2,76 @@
...
 
+# Disables generation of "version soname" (i.e. libFoo.so.<version>), which
+# causes pure duplication as part of Python wheels.
+set(CMAKE_PLATFORM_NO_VERSIONED_SONAME ON)
+
+
 # The directory at which the Python import tree begins.
 set(MPACT_PYTHON_ROOT_DIR "${CMAKE_CURRENT_SOURCE_DIR}/mpact")
 
+# We vendor our own MLIR instance in the `torch_mlir` namespace.
```
Comment (aartbik) [inline python/CMakeLists.txt:13]: torch_mlir -> mpact namespace

File: test/lit.cfg.py [hunk 2]
```diff
@@ -74,7 +74,6 @@
     "PYTHONPATH",
     [
         os.path.join(config.mpact_obj_root, "python_packages/mpact"),
-        os.path.join(config.torch_mlir_obj_root, "python_packages/torch_mlir"),
```
Comment (aartbik) [inline test/lit.cfg.py:77]: I find it strange that we can remove this now?

It looks like you moved all the torch-mlir packages into the mpact build? Can't we keep the two separate?

File: python/CMakeLists.txt [hunk 3]
```diff
@@ -2,28 +2,76 @@
...
+    MPACTCAPI
+  PRIVATE_LINK_LIBS
+    LLVMSupport
 )
 
 #-------------------------------------------------------------------------------
 # Python Modules
 #-------------------------------------------------------------------------------
 
+# To compiler TorchMLIRPythonExtension (which registers torch dialect and
```
Comment (aartbik) [inline python/CMakeLists.txt:45]: do we now package torch-mlir python into mpact?

**Review by aartbik — COMMENTED** [review]
if possible, update README

export PYTHONPATH=pwd/build/tools/mpact/python_packages/mpact

File: python/CMakeLists.txt (hunk 3 above)
Comment (PeimingLiu) [reply python/CMakeLists.txt:45]: yeah

File: python/CMakeLists.txt [hunk 4]
```diff
@@ -2,28 +2,76 @@
...
```
Comment (yinying-lisa-li) [inline python/CMakeLists.txt:45]: nit: compile instead of compiler?

**Review by aartbik — APPROVED** [review]

**Pull request merged**